src/test/groovy/steps/RepositorySteps.groovy

```diff
@@ -1,9 +1,21 @@
 package steps
 
+import tests.TestSetup
 
 import static io.restassured.RestAssured.given
 
-class RepositorySteps {
+class RepositorySteps extends TestSetup{
+
+    def acceptEula(artifactoryURL, username, password) {
+        return given()
+                .auth()
+                .preemptive()
+                .basic("${username}", "${password}")
```

Comment:
`username` and `password` are already strings. You can remove `"${}"` from them both. A `"${}"` forces groovy to make a `GString` which is _not_ the same as a Java string. This can sometimes cause incompatibilities.